src/Hive.Core/sia380/sia380.py

```diff
@@ -269,58 +278,131 @@ def main(room_properties, floor_area, T_e, setpoints_ub, setpoints_lb, surface_a
+        # Q_T, Q_V are * with eta_rec in cooling case
+        sum_QT_op_lb = sum(QT_op_per_srf_this_month_lb) # index 0 or 2
+        sum_QT_op_ub = sum(QT_op_per_srf_this_month_ub) # index 1 or 3
+        sum_QT_tr_lb = sum(QT_tr_per_srf_this_month_lb)
+        sum_QT_tr_ub = sum(QT_tr_per_srf_this_month_ub)
+
+        Q_T_list = [Q_T_lb, Q_T_ub, Q_T_lb, Q_T_ub]
+        QT_opaque_list = [sum_QT_op_lb, sum_QT_op_ub, sum_QT_op_lb, sum_QT_op_ub]
+        QT_transparent_list = [sum_QT_tr_lb, sum_QT_tr_ub, sum_QT_tr_lb, sum_QT_tr_ub]
+        Q_V_heating_list = [Q_V_lb_heating, Q_V_ub_heating, Q_V_lb_heating_no_heat_recovery, Q_V_ub_heating_no_heat_recovery]
+        Q_V_cooling_list = [Q_V_lb_cooling, Q_V_ub_cooling, Q_V_lb_cooling_no_heat_recovery, Q_V_ub_cooling_no_heat_recovery]
+        eta_rec_heating_list = [eta_g_lb_heating, eta_g_ub_heating, eta_g_lb_heating_no_hr, eta_g_ub_heating_no_hr]
+        eta_rec_cooling_list = [eta_g_lb_cooling, eta_g_ub_cooling, eta_g_lb_cooling_no_hr, eta_g_ub_cooling_no_hr]
+
+        if demand < 0:  # cooling
+            Q_T_out[month] = Q_T_list[Q_K_index] * eta_rec_cooling_list[Q_K_index]
+            QT_opaque_out[month] = QT_opaque_list[Q_K_index] * eta_rec_cooling_list[Q_K_index]
+            QT_transparent_out[month] = QT_transparent_list[Q_K_index] * eta_rec_cooling_list[Q_K_index]
+            Q_V_out[month] = Q_V_cooling_list[Q_K_index] * eta_rec_cooling_list[Q_K_index]
+            Q_i_out[month] = Q_i[month]
+            Q_s_out[month] = Q_s[month]
+        else:   # heating
+            Q_T_out[month] = Q_T_list[Q_H_index]
+            QT_opaque_out[month] = QT_opaque_list[Q_H_index]
+            QT_transparent_out[month] = QT_transparent_list[Q_H_index]
+            Q_V_out[month] = Q_V_heating_list[Q_H_index]
+            Q_i_out[month] = Q_i[month] * eta_rec_heating_list[Q_H_index]
+            Q_s_out[month] = Q_s[month] * eta_rec_heating_list[Q_H_index]
+
+
+    Q_s_tree = th.list_to_tree(Q_s_jagged, source=[0, 0])   # import ghpythonlib.treehelpers as th
 
     tokWh = 1000.0
     return [x / tokWh for x in Q_Heat], [x / tokWh for x in Q_Cool], [x / tokWh for x in Q_Elec], \
-           [x / tokWh for x in Q_T], [x / tokWh for x in Q_V], [x / tokWh for x in Q_i_eta_g], \
-           [x / tokWh for x in Q_s_eta_g], [x / tokWh for x in QT_opaque], [x / tokWh for x in QT_transparent], Q_s_tree
+           [x / tokWh for x in Q_T_out], [x / tokWh for x in Q_V_out], [x / tokWh for x in Q_i_out], \
+           [x / tokWh for x in Q_s_out], [x / tokWh for x in QT_opaque_out], [x / tokWh for x in QT_transparent_out], Q_s_tree
 
 
-def calc_eta_g(Q_T_month, Q_V_month, gamma, tau):
-    # usage of heat gains (Ausnutzungsgrad für Wärmegewinne), eta_g
-    if Q_T_month + Q_V_month < 0:
-        eta_g = 0
-    elif gamma == 1:
-        eta_g = (1 + tau / 5) / (2 + tau / 15)
+# usage of heat gains (Ausnutzungsgrad für Wärmegewinne), eta_g
+def calc_eta_g(gamma, tau, cooling):
```

Comment:
yeah. these should be two separate functions `calc_eta_g_heating` and `calc_eta_g_cooling`.

Reply:
changed it to `def calc_eta_g(gamma, tau, cooling=False):`, because I am being stubborn